### DISCUSSION TITLE
Node recovery

### DISCUSSION THREAD

**Discussion by bsenthilr** [microsoft/CCF#6669] [Q&A] [answered]
Looking for an interim option to stand minimal footprint (single node for near term). 
Can single node be recovered from snapshots to reload all committed transactions assuming uncommitted would be lost?
Assuming the service certificate would change and hence needs to be to propagated to clients similar to DR. Will the node be able to recover all transactions that where committed? 

**Reference** - [Recover sample config](https://github.com/microsoft/CCF/blob/main/samples/config/recover_config.json)

Tried curating a recovery configuration and restart the node with the old service identity. The node recovers but the "node/network" curl implies the node is waiting on `"service_status": "WaitingForRecoveryShares"`. 

`{
  "current_service_create_txid": "4.20",
  "current_view": 4,
  "primary_id": "ec2befcca7216873fc68344e398f567ac6673f13fb272272a56ffc2753fcf20d",
  "recovery_count": 1,
  "service_certificate": "*****",
  "service_data": null,
  "service_status": "WaitingForRecoveryShares"
}`

Not clear what is the right architecture and operation procedure here. Should the operations follow [Disaster Recovery](https://microsoft.github.io/CCF/main/operations/recovery.html) ?

If DR is required, should both ledger.dir & snapshot be persisted and mounted appropriately during recovery ? Does recovery use both sets of files or snapshot is sufficient?

**Comment by achamayou** [comment, marked as answer]
> Not clear what is the right architecture and operation procedure here. Should the operations follow [Disaster Recovery](https://microsoft.github.io/CCF/main/operations/recovery.html) ?

Yes

> If DR is required, should both ledger.dir & snapshot be persisted and mounted appropriately during recovery ? Does recovery use both sets of files or snapshot is sufficient?

Yes. Snapshots are unlikely to be sufficient if your goal is to avoid losing committed data, since they are produced at a configurable interval, not for every transaction. You will want at least one snapshot + ledger files since. And if you want to be able to service historical queries further back, the full ledger.

**Reply by bsenthilr** [reply]
thanks @achamayou. Let me review DR and follow up as required.
Will persist and mount both directories aptly.
